docs/book/src/cronjob-tutorial/testdata/project/cmd/main.go

```diff
@@ -114,11 +117,20 @@ func main() {
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme: scheme,
+		// Metrics endpoint is enabled in 'config/default/kustomization.yaml'. The Metrics options configure the server.
+		// For more info: https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.3/pkg/metrics/server
 		Metrics: metricsserver.Options{
-			BindAddress:    metricsAddr,
+			BindAddress:   metricsAddr,
+			SecureServing: secureMetrics,
+			// TLSOpts is used to allow configuring the TLS config used for the server. If certificates are not provided,
+			// self-signed certificates will be generated by default. This option not recommended for production
+			// environments. Replace with CertDir, CertName, and KeyName to provide certificates.
+			TLSOpts: tlsOpts,
+			// FilterProvider is used to protect the metrics endpoint with authn/authz.
+			// These configurations ensure that only authorized users and service accounts
+			// can access the metrics endpoint. The RBAC are configured in 'config/rbac/kustomization.yaml'.
+			// More info: https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.3/pkg/metrics/filters#WithAuthenticationAndAuthorization
```

Comment:
Here I just tried to clarify the options and share the recommendations
but feel free to re-word or supplement